boardswarm/src/serial/mod.rs

```diff
@@ -137,24 +137,42 @@ impl SerialPort {
         Ok(())
     }
 
-    async fn get_writer(&self) -> Arc<AsyncMutex<WriteHalf<SerialStream>>> {
+    async fn get_writer(&self) -> Result<Arc<AsyncMutex<WriteHalf<SerialStream>>>, ConsoleError> {
```

Comment:
```suggestion
   #[instrument(skip_all, err)]
    async fn get_writer(&self) -> Result<Arc<AsyncMutex<WriteHalf<SerialStream>>>, ConsoleError> {
```
This will automatically cause an error log to be triggered when the function returns an error

Reply:
I did not know this macro, I'll try it as it would make the new error handling more compact